packages/react-client/src/ReactFlightClient.js

```diff
@@ -1003,6 +1003,16 @@ function getOutlinedModel<T>(
       let value = chunk.value;
       for (let i = 1; i < path.length; i++) {
         value = value[path[i]];
+        if (value.$$typeof === REACT_LAZY_TYPE) {
+          return waitForReference(
```

Comment:
You could optimize this a bit by checking if the `_payload` chunk is already initialized and therefore just sync read it.